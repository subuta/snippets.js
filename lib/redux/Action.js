import { build, format, snippets as s } from 'bld.js'

import ActionType from './ActionType'

export default () => {
  // const imports = s.import([
  //   ['lodash', '_'],
  //   ['objection', null, [
  //     'Model'
  //   ]],
  //   ['src/utils/ajvValidator', 'ajvValidator']
  // ])

  return build`
    // action
    console.log('action')
  `
}
