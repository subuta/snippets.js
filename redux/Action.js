import { build, format, snippets as s } from 'bld.js'

export default () => {
  // const imports = s.import([
  //   ['lodash', '_'],
  //   ['objection', null, [
  //     'Model'
  //   ]],
  //   ['src/utils/ajvValidator', 'ajvValidator']
  // ])

  return build`
    // action
    console.log('action')
  `
}
