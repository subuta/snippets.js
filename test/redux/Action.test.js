/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import Action from 'lib/redux/Action'

describe('redux/Action', () => {
  it('should create redux Action', () => {
    const code = build`
      const action = ${Action('SET_BOOKS', {
        name: s.stringify('an awesome Book')
      })}
    `

    expect(format(code)).toMatchSnapshot()
  })
})
