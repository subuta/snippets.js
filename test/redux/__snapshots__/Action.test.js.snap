// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux/Action should create redux Action 1`] = `
"const action = {
  type: SET_BOOKS,
  payload: {
    name: 'an awesome Book'
  }
}
"
`;
